sputnikdao2/src/proposals.rs

```diff
@@ -570,13 +655,17 @@ impl Contract {
                 proposal.update_votes(
                     &sender_id,
                     &roles,
-                    Vote::from(action),
+                    Vote::from(action.clone()),
                     &policy,
                     self.get_user_weight(&sender_id),
                 );
+
+                // Get new status without cloning the entire proposal
+                let new_status =
+                    policy.proposal_status(&proposal, roles.clone(), self.total_delegation_amount);
+
                 // Updates proposal status with new votes using the policy.
-                proposal.status =
-                    policy.proposal_status(&proposal, roles, self.total_delegation_amount);
+                proposal.status = new_status;
```

Comment:
What is the point of introducing `new_status` if you just assign it to the `proposal.status` right after that?